src/extensions/default/CodeFolding/foldhelpers/foldcode.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * Based on http://codemirror.net/addon/fold/foldcode.js
+ * @author Patrick Oladimeji
+ * @date 10/28/13 8:41:46 AM
+ * @last modified 20 April 2014
+ */
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, brackets, document*/
+define(function (require, exports, module) {
+    "use strict";
+    var CodeMirror = brackets.getModule("thirdparty/CodeMirror2/lib/codemirror"),
+		prefs = require("Prefs");
+
+    module.exports = function () {
+        function doFold(cm, pos, options, force) {
+            force = force || "fold";
+            if (typeof pos === "number") {
+                pos = CodeMirror.Pos(pos, 0);
+            }
+            var finder = (options && options.rangeFinder) || CodeMirror.fold.auto;
```

Comment:
instead of checking is options exists in each line that uses `options`, you can have give it a default value.

``` javascript
options = options || {};
```